### PR TITLE
Fix #5928: Replace XCGLogger with os.log

### LINKED ISSUE #5928: Use os.log instead of XCGLogger

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #5928 
Security review https://github.com/brave/security/issues/1057

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: App/iOS/Delegates/AppDelegate.swift
Comment (iccub) [inline App/iOS/Delegates/AppDelegate.swift]: App/Scene delegates etc do not belong to targets? Hence the log has to be created separately in those places

File: App/iOS/Delegates/AppDelegate.swift
Comment (iccub) [inline App/iOS/Delegates/AppDelegate.swift]: Important: I set all brave core logs as public, hoping that they do not record any sensitive data.
For this case it's all or nothing, we either show full logs or make them completely private.
This is for people who can connect their devices to a Mac and run console.app

I'm ok with making it private as this does not affect us devs, all logs are visible for us when debugging.

File: Client/Frontend/Browser/BrowserViewController/BVC+Rewards.swift
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController/BVC+Rewards.swift]: We used to offer an option to export rewards logs to a file.
Since both Rewards and general code still sit in the main module, new log had to be created.

File: Client/Frontend/Settings/Rewards Internals/RewardsInternalsLogController.swift
Comment (iccub) [inline Client/Frontend/Settings/Rewards Internals/RewardsInternalsLogController.swift:46]: os.log export logic, this code is currently unused btw

File: Client/Migration/LegacyLogsMigration.swift
Comment (iccub) [inline Client/Migration/LegacyLogsMigration.swift:12]: Removing old logs

File: Client/Migration/Migration.swift
Comment (iccub) [inline Client/Migration/Migration.swift:211]: migration logic

File: Plugins/LoggerPlugin/LoggerPlugin.swift
Comment (iccub) [inline Plugins/LoggerPlugin/LoggerPlugin.swift]: I did not find a way to not hardcode the prod bundle id. Was thinking maybe it could be dynamic, but the plugin does not see Bundle.main.bundleID

**Review by iccub — COMMENTED** [review]
Marking few interesting lines of code as comments because it might be too big for many to look through

File: App/iOS/Delegates/AppDelegate.swift
Comment (cuba) [reply App/iOS/Delegates/AppDelegate.swift]: Shouldn't this be directed towards a security review?

Comment (iccub) [reply App/iOS/Delegates/AppDelegate.swift]: it will be decided on during the security review

File: Client/Frontend/Browser/Helpers/FindInPageHelper.swift
Comment (cuba) [inline Client/Frontend/Browser/Helpers/FindInPageHelper.swift]: is this potentially unsafe to spit out the whole `body`?

File: Sources/BraveShared/Analytics/UrpService.swift
Comment (cuba) [inline Sources/BraveShared/Analytics/UrpService.swift]: can you not just do `if let data = data as? Data`

File: Sources/BraveShared/Analytics/UrpService.swift
Comment (cuba) [inline Sources/BraveShared/Analytics/UrpService.swift]: same here

File: Sources/Storage/SQL/LoginsSchema.swift
Comment (cuba) [inline Sources/Storage/SQL/LoginsSchema.swift:41]: Probably can remove this `as NSError`

**Comment by cuba** [comment]
I thought that maybe its best we don't use `error.localizedDescription` (unless maybe we use `.public`) as it spits out pretty useless information. Perhaps `String(describing: error)` would be best?

**Review by cuba — APPROVED** [review]
Some small comments. Nothing that should block this PR so ill approve and let you decide on the course of action on these comments

File: Client/Frontend/Browser/Helpers/FindInPageHelper.swift
Comment (iccub) [reply Client/Frontend/Browser/Helpers/FindInPageHelper.swift]: It's a private log, not marked with `.public`

File: App/iOS/Delegates/AppDelegate.swift
Comment (stoletheminerals) [reply App/iOS/Delegates/AppDelegate.swift]: Much better to make it private imo, just in case

Comment (kylehickinson) [reply App/iOS/Delegates/AppDelegate.swift]: If its private I don't think QA will not be able view _any_ brave-core logs from Console.app… If we are going to set it private, I'd say we only set it private for Release builds

Comment (stoletheminerals) [reply App/iOS/Delegates/AppDelegate.swift]: @kylehickinson does QA team use beta builds from testflight or do they build themselves? Private for Release and public for Beta/Debug could be ok, ideally, I would make it private for the beta if possible because beta is public.

Comment (iccub) [reply App/iOS/Delegates/AppDelegate.swift]: I set it to private for now. Exposing it as public for dev/internal builds is fine to me.

Comment (iccub) [reply App/iOS/Delegates/AppDelegate.swift]: We can also make all brave core logs to never persist and be stored in-memory by forcing it to use `debug` log level.

File: Client/Frontend/Browser/TabManager.swift
Comment (stoletheminerals) [inline Client/Frontend/Browser/TabManager.swift]: Do we need this to be public?

File: Plugins/LoggerPlugin/LoggerPlugin.swift
Comment (stoletheminerals) [inline Plugins/LoggerPlugin/LoggerPlugin.swift]: Why do we dynamically generate this?

File: Sources/BraveNews/Composer/FeedDataSource.swift
Comment (stoletheminerals) [inline Sources/BraveNews/Composer/FeedDataSource.swift]: Why are you using `Darwin.log` here?

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (stoletheminerals) [inline Client/Frontend/Browser/BrowserViewController.swift]: Will this log an URL? Do we need to make it public?

File: Client/Frontend/Browser/TabManager.swift
Comment (iccub) [reply Client/Frontend/Browser/TabManager.swift]: No, only the index number is public here, not the url.
For non sensitive data it's okay to have it public, but also not a big deal for me to remove it

File: Sources/BraveNews/Composer/FeedDataSource.swift
Comment (iccub) [reply Sources/BraveNews/Composer/FeedDataSource.swift]: No changes here, in first iteration of this PR there was a name conflict, `log()` was a logging function and colluded with this method so i had to specify module.
Now logs are called via `Logger.module.debug("xx")` so I can revert this change

File: Plugins/LoggerPlugin/LoggerPlugin.swift
Comment (iccub) [reply Plugins/LoggerPlugin/LoggerPlugin.swift]: This is so every module in the app has their own logger file.
This way logs are easy to filter and categorize by their module name.
So wallet logs will have `BraveWallet` category when shown in console.app

It's an app architectural decision

File: Sources/BraveNews/Composer/FeedDataSource.swift
Comment (stoletheminerals) [reply Sources/BraveNews/Composer/FeedDataSource.swift]: I see, thanks. I thought it was a macOS-related logger for some reason

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (iccub) [reply Client/Frontend/Browser/BrowserViewController.swift]: It should print out memory address only, but makes no sense to risk it keeping public, will remove.

When I first started on this project i was deciding on whether to keep stuff public or private per each log line.
But then the changes were so big that i decided to make everything private by default and make things public as we go instead of evaluating each line

**Review by stoletheminerals — APPROVED** [review]

File: App/iOS/Delegates/AppDelegate.swift
Comment (kylehickinson) [inline App/iOS/Delegates/AppDelegate.swift]: Should we use `Bundle.main.bundleIdentifier!` here?

File: App/iOS/Delegates/AppDelegate.swift
Comment (kylehickinson) [inline App/iOS/Delegates/AppDelegate.swift]: Should we use `Bundle.main.bundleIdentifier!` here?

File: App/iOS/Delegates/CPTemplateApplicationSceneDelegate.swift
Comment (kylehickinson) [inline App/iOS/Delegates/CPTemplateApplicationSceneDelegate.swift]: Same as above

File: App/iOS/Delegates/SceneDelegate.swift
Comment (kylehickinson) [inline App/iOS/Delegates/SceneDelegate.swift]: Same as above

File: Plugins/LoggerPlugin/LoggerPlugin.swift
Comment (kylehickinson) [inline Plugins/LoggerPlugin/LoggerPlugin.swift]: There a reason to lowercasing the category here? For multiple word targets like ("BraveNews" → "bravenews") lowercasing them would make the categories less readable

File: Plugins/LoggerPlugin/LoggerPlugin.swift
Comment (kylehickinson) [inline Plugins/LoggerPlugin/LoggerPlugin.swift]: The `public` accessor here should be removed

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Just a few nits

File: Plugins/LoggerPlugin/LoggerPlugin.swift
Comment (iccub) [reply Plugins/LoggerPlugin/LoggerPlugin.swift]: No niche use where we want to call logger from another module?

Comment (kylehickinson) [reply Plugins/LoggerPlugin/LoggerPlugin.swift]: If we were to expose a public logger for the module we would want its name to be different from `module` anyways so either way we want this `module` one to be internal only

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**
